generator/nextjs/template/src/components/iota/IotaClientPage.tsx

```diff
@@ -168,88 +152,91 @@ export default function IotaSessionMultipleRequestsPage({
+  const hasErrors = !featureAvailable || !session || !session.userId;
+  const renderErrors = () => {
+    if (!featureAvailable) {
+      return (
         <div>
           Feature not available. Please set your Personal Access Token in your
           environment secrets.
         </div>
-      )}
+      );
+    }
 
-      {featureAvailable && !holderDid && (
+    if (!session || !session.userId) {
+      return (
         <div>
-          You must be logged in to share credentials from your Affinidi Vault
+          You must be logged in to issue credentials to your Affinidi Vault
         </div>
-      )}
+      );
+    }
+  };
 
-      {featureAvailable && holderDid && (
+  return (
+    <>
+      <h1 className="text-2xl font-semibold pb-6">Receive Credentials</h1>
+
+      {renderErrors()}
+      {!hasErrors && (
         <>
-          <div className="pb-4">
-            <p className="font-semibold">
-              Verified holder did (From Affinidi Login)
-            </p>
-            <p>{holderDid}</p>
-          </div>
+          {renderVerifiedHolder(session.userId)}
 
-          {configOptions.length === 0 && (
+          {configurations.isPending && (
             <div className="py-3">Loading configurations...</div>
           )}
-
-          {configOptions.length > 0 && (
+          {!configurations.isPending && (
             <Select
               id="configurationIdSelect"
               label="Configuration"
-              options={configOptions}
-              value={selectedConfig}
+              options={configurations.data || []}
+              value={selectedConfigId}
               disabled={isFormDisabled}
               onChange={handleConfigurationChange}
             />
           )}
 
-          {selectedConfig && (
+          {selectedConfigId && (
             <div>
               <Select
                 id="openModeSelect"
                 label="Open Mode"
                 options={openModeOptions}
                 value={openMode}
                 disabled={isFormDisabled}
-                onChange={handleOpenModeChange}
+                onChange={(val) => setOpenMode(val as number)}
               />
 
-              {queryOptions.length === 0 && (
+              {queryOptions.isPending && (
                 <div className="py-3">Loading queries...</div>
               )}
-              {queryOptions.length > 0 && (
+              {!queryOptions.isPending && (
                 <Select
                   id="queryId"
                   label="Query"
-                  options={queryOptions}
+                  options={queryOptions.data || []}
                   value={selectedQuery}
                   disabled={isFormDisabled}
-                  onChange={handleQueryChange}
+                  onChange={(val) => setSelectedQuery(val as string)}
                 />
               )}
 
-              {iotaSession && selectedQuery && (
+              {!iotaSession.isPending && selectedQuery && (
```

Comment:
Overall change `!xyz.isPending` with `xyz.isSuccess`

Reply:
✅ Also made better use of `.isFetching`, since `isPending` will be `true` even if the query is currently not enabled.